### ISSUE TITLE
Network resource modules fail on case insensitive file systems OSX

### ISSUE THREAD

**Issue by ganeshrn** [ansible/ansible#69350]
##### SUMMARY

* In case of nested directory structure with same names within the collection:
```
plugins/module_utils/network/eos/config/acl_interfaces/
├── __init__.py
└── acl_interfaces.py
```

* If the filename is the same at that of Python object the import from the module code can be:
```
from ansible_collections.arista.eos.plugins.module_utils.network.eos.config.acl_interfaces.acl_interfaces import Acl_interfaces
```
*  For such scenarions the collection loader logic tries to check if Acl_intertaces is a file under
   `ansible_collections.arista.eos.plugins.module_utils.network.eos.config.acl_interfaces` and since acl_interfaces.py already exist on case insensitive file syste it identifes `Acl_interfaces` as a file instead of class object within that file.
   This caused a failure for all network resource modules that follow similar design patter with error `TypeError: 'module' object is not callable\n`.

* This fix take filename case into consideration while checking if file exit or not

Collection repo: https://github.com/ansible-collections/eos
Minimal repo to reproduce the issue: https://github.com/ganeshrn/network_collections
##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
ansible 2.10.0.dev0
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

```yaml
  - name: merge attributes of given acls.
    become: true
    register: result
    arista.eos.eos_acls:
      config:
        - afi: ipv4
          acls:
            - name: test1
              aces:
                - sequence: 35
                  grant: deny
                  protocol: tcp
                  source:
                    subnet_address: 20.0.0.0/8
                  destination:
                    any: true
                  log: true
                - remark: Run by ansible
                - grant: permit
                  protocol: '6'
                  source:
                    any: true
                  destination:
                    any: true
```

##### EXPECTED RESULTS

The module should run without any error

##### ACTUAL RESULTS

Fail with error:
```paste below
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: 'module' object is not callable
fatal: [eos02]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/Users/gnalawad/.ansible/tmp/ansible-local-8192QgoxXC/ansible-tmp-1588770777.7-8203-3524252807632/AnsiballZ_eos_acls.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/Users/gnalawad/.ansible/tmp/ansible-local-8192QgoxXC/ansible-tmp-1588770777.7-8203-3524252807632/AnsiballZ_eos_acls.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/Users/gnalawad/.ansible/tmp/ansible-local-8192QgoxXC/ansible-tmp-1588770777.7-8203-3524252807632/AnsiballZ_eos_acls.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.arista.eos.plugins.modules.eos_acls', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py\", line 188, in run_module\n    fname, loader, pkg_name)\n  File \"/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py\", line 82, in _run_module_code\n    mod_name, mod_fname, mod_loader, pkg_name)\n  File \"/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py\", line 72, in _run_code\n    exec code in run_globals\n  File \"/var/folders/8l/27cd4l3n2vnc2cpk3498__680000gp/T/ansible_arista.eos.eos_acls_payload_jxsk65/ansible_arista.eos.eos_acls_payload.zip/ansible_collections/arista/eos/plugins/modules/eos_acls.py\", line 914, in <module>\n  File \"/var/folders/8l/27cd4l3n2vnc2cpk3498__680000gp/T/ansible_arista.eos.eos_acls_payload_jxsk65/ansible_arista.eos.eos_acls_payload.zip/ansible_collections/arista/eos/plugins/modules/eos_acls.py\", line 909, in main\nTypeError: 'module' object is not callable\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Comment by ganeshrn** [comment]
PR https://github.com/ansible/ansible/pull/67817 is a workaround to unblock the development process and not a final fix

**Comment by justjais** [comment]
Any update on the fix for the issue as currently, I am managing the failures by manually updating the code applying the workaround patch fix changes.

**Comment by ganeshrn** [comment]
The fix for this issue needs to be backported to stable-2.9 branch and is a blocker for supported collections 1.0.0 release

**Comment by ganeshrn** [comment]
Ansible version:
```
$ansible --version
ansible 2.9.9.post0
  config file = /Users/gnalawad/Documents/ganeshrn/ansible-collections/ansible.cfg
  configured module search path = ['/Users/gnalawad/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/gnalawad/Documents/ganeshrn/ansible/lib/ansible
  executable location = /Users/gnalawad/Documents/ganeshrn/ansible/bin/ansible
  python version = 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21) [Clang 6.0 (clang-600.0.57)]

```

OS / ENVIRONMENT
```
$sw_vers
ProductName:    Mac OS X
ProductVersion: 10.14.6
BuildVersion:   18G4032

```
STEPS TO REPRODUCE:

1) Install vyos collection from galaxy
```
ansible-galaxy collection install vyos.vyos
```
2) Run playbook:
state: rendered doesn't require active connection with remote host.
```
---
- hosts: vyos
  gather_facts: no
  connection: ansible.netcommon.network_cli
  tasks:
  - name: Render the commands for provided  configuration
    vyos.vyos.vyos_interfaces:
      config:
      - name: eth0
        enabled: true
        duplex: auto
        speed: auto
      state: rendered
```
EXPECTED RESULTS
The module should run without any error
```
ok: [vyos01] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "config": [
                {
                    "description": null,
                    "duplex": "auto",
                    "enabled": true,
                    "mtu": null,
                    "name": "eth0",
                    "speed": "auto",
                    "vifs": null
                }
            ],
            "running_config": null,
            "state": "rendered"
        }
    },
    "rendered": [
        "delete interfaces ethernet eth0 disable",
        "set interfaces ethernet eth0 duplex 'auto'",
        "set interfaces ethernet eth0 speed 'auto'"
    ]
}

```
ACTUAL RESULTS
Fail with error:
```
fatal: [vyos01]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/Users/gnalawad/.ansible/tmp/ansible-local-97675oe3njdn5/ansible-tmp-1591333989.189882-97681-38408817799145/AnsiballZ_vyos_interfaces.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/Users/gnalawad/.ansible/tmp/ansible-local-97675oe3njdn5/ansible-tmp-1591333989.189882-97681-38408817799145/AnsiballZ_vyos_interfaces.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/Users/gnalawad/.ansible/tmp/ansible-local-97675oe3njdn5/ansible-tmp-1591333989.189882-97681-38408817799145/AnsiballZ_vyos_interfaces.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.vyos.vyos.plugins.modules.vyos_interfaces', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/var/folders/8l/27cd4l3n2vnc2cpk3498__680000gp/T/ansible_vyos.vyos.vyos_interfaces_payload_nx23kauh/ansible_vyos.vyos.vyos_interfaces_payload.zip/ansible_collections/vyos/vyos/plugins/modules/vyos_interfaces.py\", line 1131, in <module>\n  File \"/var/folders/8l/27cd4l3n2vnc2cpk3498__680000gp/T/ansible_vyos.vyos.vyos_interfaces_payload_nx23kauh/ansible_vyos.vyos.vyos_interfaces_payload.zip/ansible_collections/vyos/vyos/plugins/modules/vyos_interfaces.py\", line 1126, in main\nTypeError: 'module' object is not callable\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

```

**Comment by nitzmahone** [comment]
@justjais can you validate this fix? It fixes the repro @ganeshrn gave me on Mac Python 2.7. Assuming it works, tests pass, etc, this should be included in the 2.9.10 release in the next week or so.

**Comment by justjais** [comment]
@nitzmahone yes, it's working as expected with the fix, thank you. @ganeshrn As the issue is fixed both in `devel` and `stable-2.9` going ahead and closing the issue.
